tests/cp.go

```diff
@@ -0,0 +1,147 @@
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				command.Run(o, "cp", containerResource, path)
+				fileShouldExist(path, content)
+			})
+
+			for _, link := range []string{"-L", "--follow-link"} {
+				ginkgo.It(fmt.Sprintf("with %s flag, should be able to copy file from host to container and follow symbolic link",
+					link), func() {
+					path := ffs.CreateTempFile(filename, content)
+					fileDir := filepath.Dir(path)
+					ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+					symlink := filepath.Join(fileDir, "symlink")
+					err := os.Symlink(path, symlink)
+					gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+					command.Run(o, "cp", link, symlink, containerResource)
+					fileShouldExistInContainer(o, testContainerName, containerFilepath, content)
+				})
+
+				ginkgo.It(fmt.Sprintf("with %s flag, should be able to copy file from container to host and follow symbolic link",
+					link), func() {
+					cmd := fmt.Sprintf("echo -n %s > %s", content, containerFilepath)
+					command.Run(o, "exec", testContainerName, "sh", "-c", cmd)
+					containerSymlink := filepath.Join("/tmp", "symlink")
+					command.Run(o, "exec", testContainerName, "ln", "-s", containerFilepath, containerSymlink)
+					fileDir := ffs.CreateTempDir("finch-test")
+					path := filepath.Join(fileDir, filename)
+					ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+					command.Run(o, "cp", link, fmt.Sprintf("%s:%s", testContainerName, containerSymlink), path)
+					fileShouldExist(path, content)
+				})
+			}
+
+			ginkgo.It("should not be able to copy nonexistent file from host to container", func() {
+				fileDir := ffs.CreateTempDir("finch-test")
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", filepath.Join(fileDir, filename), containerResource)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("no such file or directory"))
+				fileShouldNotExistInContainer(o, testContainerName, containerFilepath)
+			})
+
+			ginkgo.It("should not be able to copy nonexistent file from container to host", func() {
+				fileDir := ffs.CreateTempDir("finch-test")
+				path := filepath.Join(fileDir, filename)
+				ginkgo.DeferCleanup(os.RemoveAll, fileDir)
+
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", containerResource, path)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("no such file or directory"))
+				fileShouldNotExist(path)
+			})
+		})
+
+		ginkgo.Context("when the container is not running", func() {
+			ginkgo.It("should not be able to copy file from host to container", func() {
+				command.Run(o, "run", "--name", testContainerName, defaultImage)
+				path := ffs.CreateTempFile(filename, content)
+				ginkgo.DeferCleanup(os.RemoveAll, filepath.Dir(path))
+				cmdOut := command.RunWithoutSuccessfulExit(o, "cp", path, containerResource)
+				gomega.Expect(cmdOut.Err.Contents()).To(gomega.ContainSubstring("expected container status running"))
```

Comment:
Currently if we want to verify the text output, we only verify something we can control. These texts are from upstream and we can't easily control what does it look like. I think `command.RunWithoutSuccessfulExit(o, "cp", path, containerResource)` should be enough. Suggest removing output validation from upstream for now.

Reply:
Makes sense. I will also remove the checks for `finch cp` failure cases.